osu.Game/Rulesets/Edit/HitObjectComposer.cs

```diff
@@ -242,7 +247,7 @@ protected override void Update()
             base.Update();
 
             // Ensure that the playfield is always centered but also doesn't get cut off by toolboxes.
-            PlayfieldContentContainer.Width = Math.Max(1024, DrawWidth) - TOOLBOX_CONTRACTED_SIZE_RIGHT * 2;
+            PlayfieldContentContainer.Width = Math.Max(1024, DrawWidth) - (ApplyVerticalCentering ? TOOLBOX_CONTRACTED_SIZE_RIGHT : TOOLBOX_CONTRACTED_SIZE_LEFT) * 2;
```

Comment:
And secondly, this application method means that the right edge of the playfield may get cut off again. This is because `PlayfieldContentContainer` is anchored to centre, and what this essentially does is add a padding on the side of either 60 if the flag is true, or 120 if the flag is false. But that means that if the flag is true, then potentially up to 30 units on the right side may get cut off by the right toolbox.

The more correct way to do this would probably something like

```diff
diff --git a/osu.Game/Rulesets/Edit/HitObjectComposer.cs b/osu.Game/Rulesets/Edit/HitObjectComposer.cs
index 7612d8b161..0cd3436bb1 100644
--- a/osu.Game/Rulesets/Edit/HitObjectComposer.cs
+++ b/osu.Game/Rulesets/Edit/HitObjectComposer.cs
@@ -247,7 +247,16 @@ protected override void Update()
             base.Update();
 
             // Ensure that the playfield is always centered but also doesn't get cut off by toolboxes.
-            PlayfieldContentContainer.Width = Math.Max(1024, DrawWidth) - (ApplyVerticalCentering ? TOOLBOX_CONTRACTED_SIZE_RIGHT : TOOLBOX_CONTRACTED_SIZE_LEFT) * 2;
+            if (ApplyVerticalCentering)
+            {
+                PlayfieldContentContainer.Width = Math.Max(1024, DrawWidth) - TOOLBOX_CONTRACTED_SIZE_RIGHT * 2;
+                PlayfieldContentContainer.X = 0;
+            }
+            else
+            {
+                PlayfieldContentContainer.Width = Math.Max(1024, DrawWidth) - (TOOLBOX_CONTRACTED_SIZE_LEFT + TOOLBOX_CONTRACTED_SIZE_RIGHT);
+                PlayfieldContentContainer.X = (TOOLBOX_CONTRACTED_SIZE_LEFT - TOOLBOX_CONTRACTED_SIZE_RIGHT) / 2;
+            }
         }
 
         public override Playfield Playfield => drawableRulesetWrapper.Playfield;

```

which can be tested by turning off centering elsewhere and examining osu!, for instance. With the flag off and no further changes it'll be cut off; with the flag off and changes above it'll fit perfectly.

Reply:
I was aware of this but couldn't see any immediate repercussions. Maybe more relevant for other rulesets

Reply:
I've applied what I think is a slightly more correct version of your fix. With your method, you were still centering based on the average of the toolboxes, which might not work in all cases. Or maybe it would, but it doesn't read clearly and I don't have the brainpower to math that, so I think using notation we're more familiar with makes sense here.